src/csync/csync_update.cpp

```diff
@@ -245,57 +208,57 @@ static int _csync_detect_update(CSYNC *ctx, const char *file,
 
             // Checksum comparison at this stage is only enabled for .eml files,
             // check #4754 #4755
-            bool isEmlFile = csync_fnmatch("*.eml", file, FNM_CASEFOLD) == 0;
+            bool isEmlFile = csync_fnmatch("*.eml", fs->path, FNM_CASEFOLD) == 0;
             if (isEmlFile && fs->size == tmp->size && !tmp->checksumHeader.isEmpty()) {
                 if (ctx->callbacks.checksum_hook) {
-                    st->checksumHeader = ctx->callbacks.checksum_hook(
-                        file, tmp->checksumHeader,
+                    fs->checksumHeader = ctx->callbacks.checksum_hook(
+                        _rel_to_abs(ctx, fs->path), tmp->checksumHeader,
```

Comment:
... so we don't need to allocate memory here for the path.

Reply:
This is currently only for `.eml` files, and I don't like having a mix of absolute and relative paths here and there. I think always using absolute makes things easier to follow.

We're already doing tons of allocations everywhere, I think that having one here is fine, and I'm already saving quite a few by passing CoW QByteArrays instead of c_strdups everywhere.
Let me know what you think.